SmartDeviceLink/private/SDLTextAndGraphicUpdateOperation.m

```diff
@@ -516,29 +516,45 @@ - (void)sdl_updateCurrentScreenDataFromSetDisplayLayout:(SDLSetDisplayLayout *)s
 #pragma mark - Should Update
 
 - (BOOL)sdl_artworkNeedsUpload:(SDLArtwork *)artwork {
-    return (artwork != nil && ![self.fileManager hasUploadedFile:artwork] && !artwork.isStaticIcon);
+    if (artwork != nil) {
+        if (artwork.isStaticIcon) {
+            return NO;
+        } else {
+            return (artwork.overwrite || (![self.fileManager hasUploadedFile:artwork]));
+        }
+    }
+    return NO;
+}
+
+- (BOOL) sdl_artworkUploaded:(SDLArtwork *)artwork {
+    if (artwork != nil) {
+        return artwork.isStaticIcon || (self.fileManager != nil && [self.fileManager hasUploadedFile:artwork]);
+    }
+    return YES;
```

Comment:
```suggestion
    return NO;
```

If the artwork is `nil`, it's not uploaded, right? Note, this may break some things, so it may need a refactor. Returning `YES` just doesn't make sense here.